app/src/main/java/com/github/polypoly/app/base/game/rules_and_lobby/GameLobbyActivity.kt

```diff
@@ -0,0 +1,193 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonColors
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.github.polypoly.app.base.game.rules_and_lobby.kotlin.GameLobby
+import com.github.polypoly.app.base.user.User
+import com.github.polypoly.app.global.GlobalInstances.Companion.remoteDB
+import com.github.polypoly.app.global.Settings.Companion.DB_GAME_LOBBIES_PATH
+import com.github.polypoly.app.map.MapActivity
+import com.github.polypoly.app.ui.theme.PolypolyTheme
+import com.github.polypoly.app.utils.Padding
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.ValueEventListener
+
+class GameLobbyActivity : ComponentActivity() {
+    private lateinit var gameLobby: MutableState<GameLobby>
+
+    private val playerPerRow = 3
+
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        receiveGameLobby()
+        setContent { GameLobbyContent() }
+    }
+
+    @Preview(showBackground = true)
+    @Composable
+    private fun GameLobbyPreview() { GameLobbyContent() }
+
+    // ===================================================== MAIN CONTENT
+    @Composable
+    private fun GameLobbyContent() {
+        gameLobby = remember { mutableStateOf(GameLobby()) }
+        PolypolyTheme {
+            Surface(
+                modifier = Modifier.fillMaxSize(),
+                color = MaterialTheme.colors.background
+            ) {
+                Column(
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    Text("Game Lobby name: ${gameLobby.value.name}")
+                    Button(onClick = { /*TODO*/ }) {
+                        Text(text = "QUIT")
+                    }
+                    PlayerGrid()
+                    GoButton()
+                }
+            }
+        }
+
+    }
+
+    // ===================================================== GAME LOBBY COMPONENTS
+
+    @Composable
+    private fun PlayerGrid() {
+        val numberOfPlayers = gameLobby.value.usersRegistered.size
+        val numberOfRows: Int = numberOfPlayers / playerPerRow
```

Comment:
What happens if the division result is not an exact integer? One row less will be counted because of the rounding! The integer division in kotlin is the same of java.

Reply:
it works :)